File: software/server/src/main/java/br/labcin/sade/modelo/LesaoCirurgia.java
package br.labcin.sade.modelo;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Arrays;
import java.util.Date;
import java.util.HashMap;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.fasterxml.jackson.annotation.JsonIgnore;


@Entity
public class LesaoCirurgia {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private boolean auditado; // NOVO	
		
	@Column(nullable=false)
	private String diagnosticoClinico;
		
	private String diagnosticoClinicoSecundario; // NOVO
	
	@Column(nullable = false)
	private String regiao;
	
	private String grandeRegiao;
	
	@Column(nullable = false)
	private String procedimento;
	
	private String cirurgiao;
	
	@Column(length=500)
	private String obs;
	
	@Column(nullable = false)
	private float diametroMaior;
	
	@Column(nullable = false)
	private float diametroMenor;	
	
	@ManyToOne
	@JoinColumn(name="paciente_id")
	@JsonIgnore 
	private PacienteCirurgia paciente;	
	
	@OneToMany	(fetch = FetchType.LAZY, cascade=CascadeType.ALL, mappedBy="lesao")	
	private List<ImagemCirurgia> imagens;
	
	@Temporal(TemporalType.DATE)
	// @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", locale = "pt-BR", timezone = "Brazil/East")
	@Column(nullable = false)
	private Date dataProcedimento; // MUDOU
	
	@Column(nullable = false)
	private String localProcedimento; // MUDOU
	
	// Dados a serem preenchido quando o histopatologico for realizado
	private String diagnosticoHisto;
		
	private String subtipoHisto;
		
	private String clark;
	
	private float breslow;
	
	private float indiceMiotico;

	private String tipoTumor;
	
    private float tamanhoTumorDimMaior;

	private float tamanhoTumorDimMenor;
    
    private String crescimentoRadial;
    
    private String crescimentoVertical;
    
    private String ulceracao;

	private String lesaoSatelite;
    
    private String evidenciaRegressao;
    
    private String associacaoNevoMelanocito;
    
    private String margemCirurgiaProfunda;
    
    private String margemCirurgiaLateral;
    
    private String margemProfundaLivre;    

	private String margemLateralLivre;
    
    private String invasaoVascular;
    
    private String menorDistMargemRessecaoLateral;
    
    private String limiteRessecao;
    
    private String infiltracaoPerineural;
    
    private String infiltracaoAngiolinfatica;
    
    private String infiltradoLinfocito;
    

    

	
	//##############################################################################################
	// IMPLEMENTAÇÃO DE MÉTODOS DA CLASSE
	//##############################################################################################	
	
	public void print() {
		System.out.println("---- LESAO ----");
		System.out.println("ID: " + id);
		System.out.println("Regiao: " + regiao);
		System.out.println("Diagnostico Clinico: " + diagnosticoClinico);
		System.out.println("Diagnostico Histopatologico: " + diagnosticoHisto);
		System.out.println("Diametro maior: " + diametroMaior);
		System.out.println("Diametro menor: " + diametroMenor);
		System.out.println("Procedimento: " + procedimento);
		System.out.println("Cirurgião: " + cirurgiao);
		
		
		System.out.println("--- IMAGENS ---");
		if (!imagens.isEmpty()) {
			for (ImagemCirurgia img : imagens) {
				img.print();
			}
		}
		System.out.println("-------------------------------------------\n");
		 
	}

	public String datasetLinha (HashMap<String, String> pacienteStr, String[] dadosReq) {
		String linha = "";
		HashMap<String, String> lesaoStr = new HashMap<String, String>();
		DateFormat dateFormat = new SimpleDateFormat("dd/mm/yyyy");

		for (String dado: dadosReq) {
			switch (dado) {
				case "diagnosticoClinico":
					lesaoStr.put( dado, diagnosticoClinico );
					break;
				case "diagnosticoClinicoSecundario":
					lesaoStr.put( dado, diagnosticoClinicoSecundario );
					break;
				case "regiao":
					lesaoStr.put( dado, regiao );
					break;
				case "grandeRegiao":
					lesaoStr.put( dado, grandeRegiao );
					break;
				case "procedimento":
					lesaoStr.put( dado, procedimento );
					break;
				case "cirurgiao":
					lesaoStr.put( dado, cirurgiao );
					break;
				case "obs":
					lesaoStr.put( dado, obs );
					break;
				case "diametroMaior":
					lesaoStr.put( dado, String.valueOf(diametroMaior) );
					break;
				case "diametroMenor":
					lesaoStr.put( dado, String.valueOf(diametroMenor) );
					break;
				case "dataProcedimento":
					lesaoStr.put( dado, dateFormat.format(dataProcedimento) );
					break;
				case "localProcedimento":
					lesaoStr.put( dado, localProcedimento );
					break;
				case "diagnosticoHisto":
					lesaoStr.put( dado, diagnosticoHisto );
					break;
				case "subtipoHisto":
					lesaoStr.put( dado, subtipoHisto );
					break;
				case "clark":
					lesaoStr.put( dado, clark );
					break;
				case "breslow":
					lesaoStr.put( dado, String.valueOf(breslow) );
					break;
				case "indiceMiotico":
					lesaoStr.put( dado, String.valueOf(indiceMiotico) );
					break;
				case "tipoTumor":
					lesaoStr.put( dado, tipoTumor );
					break;
				case "tamanhoTumorDimMaior":
					lesaoStr.put( dado, String.valueOf(tamanhoTumorDimMaior) );
					break;
				case "tamanhoTumorDimMenor":
					lesaoStr.put( dado, String.valueOf(tamanhoTumorDimMenor) );
					break;
				case "crescimentoRadial":
					lesaoStr.put( dado, crescimentoRadial );
					break;
				case "crescimentoVertical":
					lesaoStr.put( dado, crescimentoVertical );
					break;
				case "ulceracao":
					lesaoStr.put( dado, ulceracao );
					break;
				case "lesaoSatelite":
					lesaoStr.put( dado, lesaoSatelite );
					break;
				case "evidenciaRegressao":
					lesaoStr.put( dado, evidenciaRegressao );
					break;
				case "associacaoNevoMelanocito":
					lesaoStr.put( dado, associacaoNevoMelanocito );
					break;
				case "margemCirurgiaProfunda":
					lesaoStr.put( dado, margemCirurgiaProfunda );
					break;
				case "margemCirurgiaLateral":
					lesaoStr.put( dado, margemCirurgiaLateral );
					break;
				case "margemProfundaLivre":
					lesaoStr.put( dado, margemProfundaLivre );
					break;
				case "margemLateralLivre":
					lesaoStr.put( dado, margemLateralLivre );
					break;
				case "invasaoVascular":
					lesaoStr.put( dado, invasaoVascular );
					break;
				case "menorDistMargemRessecaoLateral":
					lesaoStr.put( dado, menorDistMargemRessecaoLateral );
					break;
				case "limiteRessecao":
					lesaoStr.put( dado, limiteRessecao );
					break;
				case "infiltracaoPerineural":
					lesaoStr.put( dado, infiltracaoPerineural );
					break;
				case "infiltracaoAngiolinfatica":
					lesaoStr.put( dado, infiltracaoAngiolinfatica );
					break;
				case "infiltradoLinfocito":
					lesaoStr.put( dado, infiltradoLinfocito );
					break;
				default:
					break;
			}
		}

		HashMap<String, String> all = new HashMap<String, String>(pacienteStr);
		all.putAll(lesaoStr);
		String allStr = "";
		for (String dado: dadosReq) {
			allStr += all.get(dado) + ';';
		}

		if(Arrays.stream(dadosReq).anyMatch("path"::equals)) {
			for (ImagemCirurgia img : imagens) {
				linha += allStr + img.getPath() + '\n';
			}
		} else {
			linha = allStr.replaceFirst("(?s);(?!.*?;)", "\n");
		}
		return linha;
	}
	//##############################################################################################
	// MÉTODOS DA CLASSE
	//##############################################################################################	
	
	public long contaImagens () {
		return this.imagens.size();
	}
	
	//##############################################################################################
	// INICIO DOS GETS E SETS
	//##############################################################################################

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getDiagnosticoClinico() {
		return diagnosticoClinico;
	}

	public void setDiagnosticoClinico(String diagnosticoClinico) {
		this.diagnosticoClinico = diagnosticoClinico.toUpperCase();
	}

	public String getDiagnosticoHisto() {
		return diagnosticoHisto;
	}

	public void setDiagnosticoHisto(String diagnosticoHisto) {
		if(diagnosticoHisto == null) {
			this.diagnosticoHisto = diagnosticoHisto;
		}else {
			this.diagnosticoHisto = diagnosticoHisto.toUpperCase();
		}
		
	}

	public String getSubtipoHisto() {
		return subtipoHisto;
	}

	public void setSubtipoHisto(String subtipoHisto) {
		
		if(subtipoHisto == null) {
			this.subtipoHisto = subtipoHisto;
		}else {
			this.subtipoHisto = subtipoHisto.toUpperCase();
		}
	}

	public String getClark() {
		return clark;
	}

	public void setClark(String clark) {
		if (clark == null) {
			this.clark = clark;
		} else {
			this.clark = clark.toUpperCase();
		}
	}

	public float getBreslow() {
		return breslow;
	}

	public void setBreslow(float breslow) {
		this.breslow = breslow;
	}

	public float getIndiceMiotico() {
		return indiceMiotico;
	}

	public void setIndiceMiotico(float indiceMiotico) {
		this.indiceMiotico = indiceMiotico;
	}

	public String getRegiao() {
		return regiao;
	}

	public void setRegiao(String regiao) {
		this.regiao = regiao.toUpperCase();
	}

	public String getProcedimento() {
		return procedimento;
	}

	public void setProcedimento(String procedimento) {
		this.procedimento = procedimento.toUpperCase();
	}

	public String getCirurgiao() {
		return cirurgiao;
	}

	public void setCirurgiao(String cirurgiao) {
		this.cirurgiao = cirurgiao.toUpperCase();
	}

	public String getObs() {
		return obs;
	}

	public void setObs(String obs) {
		this.obs = obs.toUpperCase();
	}

	public float getDiametroMaior() {
		return diametroMaior;
	}

	public void setDiametroMaior(float diametroMaior) {
		this.diametroMaior = diametroMaior;
	}

	public float getDiametroMenor() {
		return diametroMenor;
	}

	public void setDiametroMenor(float diametroMenor) {
		this.diametroMenor = diametroMenor;
	}

	public List<ImagemCirurgia> getImagens() {
		return imagens;
	}

	public void setImagens(List<ImagemCirurgia> imagens) {
		this.imagens = imagens;
	}

	public Date getDataProcedimento() {
		return dataProcedimento;
	}

	public void setDataProcedimento(Date dataProcedimento) {
		this.dataProcedimento = dataProcedimento;
	}

	public String getLocalProcedimento() {
		return localProcedimento;
	}

	public void setLocalProcedimento(String localProcedimento) {
		this.localProcedimento = localProcedimento.toUpperCase();
	}

	public String getDiagnosticoClinicoSecundario() {
		return diagnosticoClinicoSecundario;
	}

	public void setDiagnosticoClinicoSecundario(String diagnosticoClinicoSecundario) {
		this.diagnosticoClinicoSecundario = diagnosticoClinicoSecundario.toUpperCase();
	}

	public PacienteCirurgia getPaciente() {
		return paciente;
	}

	public void setPaciente(PacienteCirurgia paciente) {
		this.paciente = paciente;
	}

	public boolean isAuditado() {
		return auditado;
	}

	public void setAuditado(boolean auditado) {
		this.auditado = auditado;
	}
	
	public String getTipoTumor() {
        return tipoTumor;
    }

    public void setTipoTumor(String tipoTumor){
    	if (tipoTumor == null) {
    		this.tipoTumor = tipoTumor;
    	}else {
    		this.tipoTumor = tipoTumor.toUpperCase();
    	}
        
    }

    public String getCrescimentoRadial() {
        return crescimentoRadial;
    }

    public void setCrescimentoRadial(String crescimentoRadial){
    	if (crescimentoRadial == null) {
    		this.crescimentoRadial = crescimentoRadial;
    	} else {
    		this.crescimentoRadial = crescimentoRadial.toUpperCase();
    	}
        
    }

    public String getCrescimentoVertical() {
        return crescimentoVertical;
    }

    public void setCrescimentoVertical(String crescimentoVertical) {
    	if(crescimentoVertical == null) {
    		this.crescimentoVertical = crescimentoVertical;
    	} else {
    		this.crescimentoVertical = crescimentoVertical.toUpperCase();
    	}
        
    }

    public String  getUlceracao() {
        return ulceracao;
    }

    public void setUlceracao(String ulceracao) {
    	if (ulceracao == null) {
    		this.ulceracao = ulceracao;
    	} else {
    		this.ulceracao = ulceracao.toUpperCase();
    	}
        
    }

    public String getLesaoSatelite() {
        return lesaoSatelite;
    }

    public void setLesaoSatelite(String lesaoSatelite) {
    	if (lesaoSatelite == null) {
    		this.lesaoSatelite = lesaoSatelite;
    	} else {
    		this.lesaoSatelite = lesaoSatelite.toUpperCase();
    	}
        
    }

    public String getEvidenciaRegressao() {
        return evidenciaRegressao;
    }

    public void setEvidenciaRegressao(String evidenciaRegressao) {
    	if (evidenciaRegressao == null) {
    		this.evidenciaRegressao = evidenciaRegressao;
    	} else {
    		this.evidenciaRegressao = evidenciaRegressao.toUpperCase();
    	}
        
    }

    public String getAssociacaoNevoMelanocito() {
        return this.associacaoNevoMelanocito;
    }

    public void setAssociacaoNevoMelanocito(String associacaoNevoMelanocito) {
    	if (associacaoNevoMelanocito == null) {
    		this.associacaoNevoMelanocito = associacaoNevoMelanocito;
    	} else {
    		this.associacaoNevoMelanocito = associacaoNevoMelanocito.toUpperCase();
    	}
        
    }

    public String getMargemCirurgiaProfunda() {
        return margemCirurgiaProfunda;
    }

    public void setMargemCirurgiaProfunda(String margemCirurgiaProfunda) {
    	if (margemCirurgiaProfunda == null) {
    		this.margemCirurgiaProfunda = margemCirurgiaProfunda;
    	} else {
    		 this.margemCirurgiaProfunda = margemCirurgiaProfunda.toUpperCase();
    	}
       
    }

    public String getMargemCirurgiaLateral() {
        return margemCirurgiaLateral;
    }

    public void setMargemCirurgiaLateral(String margemCirurgiaLateral) {
    	if (margemCirurgiaLateral == null) {
    		this.margemCirurgiaLateral = margemCirurgiaLateral;
    	}else {
    		this.margemCirurgiaLateral = margemCirurgiaLateral.toUpperCase();
    	}
        
    }

    public String getInvasaoVascular() {
        return invasaoVascular;
    }

    public void setInvasaoVascular(String invasaoVascular) {
    	if (invasaoVascular == null) {
    		this.invasaoVascular = invasaoVascular;
    	}else {
    		 this.invasaoVascular = invasaoVascular.toUpperCase();
    	}
       
    }

    public String getMenorDistMargemRessecaoLateral() {
        return menorDistMargemRessecaoLateral;
    }

    public void setMenorDistMargemRessecaoLateral(String menorDistMargemRessecaoLateral) {
    	if( menorDistMargemRessecaoLateral == null) {
    		this.menorDistMargemRessecaoLateral = menorDistMargemRessecaoLateral;
    	}else {
    		 this.menorDistMargemRessecaoLateral = menorDistMargemRessecaoLateral.toUpperCase();
    	}
    	
    }
    
    public String getLimiteRessecao() {
		return limiteRessecao;
	}

	public void setLimiteRessecao(String limiteRessecao) {
		if (limiteRessecao == null) {
			this.limiteRessecao = limiteRessecao;
		} else {
			this.limiteRessecao = limiteRessecao.toUpperCase();
		}
		
	}

	public String getInfiltracaoPerineural() {
		return infiltracaoPerineural;
	}

	public void setInfiltracaoPerineural(String infiltracaoPerineural) {
		if (infiltracaoPerineural == null) {
			this.infiltracaoPerineural = infiltracaoPerineural;
		} else {
			this.infiltracaoPerineural = infiltracaoPerineural.toUpperCase();
		}
		
		
	}

	public String getInfiltracaoAngiolinfatica() {
		return infiltracaoAngiolinfatica;
	}

	public void setInfiltracaoAngiolinfatica(String infiltracaoAngiolinfatica) {
		if (infiltracaoAngiolinfatica == null) {
			this.infiltracaoAngiolinfatica = infiltracaoAngiolinfatica;
		} else {
			this.infiltracaoAngiolinfatica = infiltracaoAngiolinfatica.toUpperCase();
		}
		
	}

	public String getInfiltradoLinfocito() {
		return infiltradoLinfocito;
	}

	public void setInfiltradoLinfocito(String infiltradoLinfocito) {
		if(infiltradoLinfocito == null) {
			this.infiltradoLinfocito = infiltradoLinfocito;
		} else {
			this.infiltradoLinfocito = infiltradoLinfocito.toUpperCase();
		}
		
	}
	
	public float getTamanhoTumorDimMaior() {
		return tamanhoTumorDimMaior;
	}

	public void setTamanhoTumorDimMaior(float tamanhoTumorDimMaior) {
		this.tamanhoTumorDimMaior = tamanhoTumorDimMaior;
	}

	public float getTamanhoTumorDimMenor() {
		return tamanhoTumorDimMenor;
	}

	public void setTamanhoTumorDimMenor(float tamanhoTumorDimMenor) {
		this.tamanhoTumorDimMenor = tamanhoTumorDimMenor;
	}
	
	public String getMargemProfundaLivre() {
		return margemProfundaLivre;
	}

	public void setMargemProfundaLivre(String margemProfundaLivre) {
		if(margemProfundaLivre == null) {
			this.margemProfundaLivre = margemProfundaLivre;
		}else {
			this.margemProfundaLivre = margemProfundaLivre.toUpperCase();
		}
		
	}

	public String getMargemLateralLivre() {
		return margemLateralLivre;
	}

	public void setMargemLateralLivre(String margemLateralLivre) {
		if (margemLateralLivre == null) {
			this.margemLateralLivre = margemLateralLivre;
		}else {
			this.margemLateralLivre = margemLateralLivre.toUpperCase();
		}
		
	}

	public String getGrandeRegiao() {
		return grandeRegiao;
	}

	public void setGrandeRegiao(String grandeRegiao) {
		this.grandeRegiao = grandeRegiao;
	}	
	
}

File: software/frontend/src/app/modelo/LesaoCirurgia.ts
import Imagem from './Imagem';

export default class LesaoCirurgia {

    // Dados gerais da lesão do paciente
    private id: number;
    private diagnosticoClinico: string;
    private diagnosticoClinicoSecundario: string;
    private procedimento: string;
    private cirurgiao: string;
    private obs: string;
    private diametroMaior: number;
    private diametroMenor: number;
    private imagens: Array<Imagem>;
    private dataProcedimento: Date;
    private localProcedimento: string;
    private regiao: string;
    private grandeRegiao: string;

    private pacienteId: string;

    // Dados para completar quando tiver o histopatologico
    private diagnosticoHisto: string;
    private subtipoHisto: string;
    private clark: string;
    private breslow: number;
    private indiceMiotico: number;
    private tipoTumor: string;
    private tamanhoTumorDimMaior: number;
    private tamanhoTumorDimMenor: number;
    private crescimentoRadial: string;
    private crescimentoVertical: string;
    private ulceracao: string;
    private lesaoSatelite: string;
    private evidenciaRegressao: string;
    private associacaoNevoMelanocito: string;
    private margemCirurgiaProfunda: string;
    private margemCirurgiaLateral: string;
    private margemLateralLivre: string; // infs extra para margem lateral livre
    private margemProfundaLivre: string; // infs extra para margem profunda livre
    private invasaoVascular: string;
    private menorDistMargemRessecaoLateral: string;
    private limiteRessecao: string;
    private infiltracaoPerineural: string;
    private infiltracaoAngiolinfatica: string;
    private infiltradoLinfocito: string;
    private auditado: boolean;


    constructor (dados?: any) {

        // Toda lesao adicionana precisa ser auditada
        this.auditado = false;

        if (dados !== undefined) {

            // o id só é preenchido qnd o paciente é retornado do banco
            this.id = dados.id;
            this.diagnosticoClinico = dados.diagnosticoClinico.toUpperCase();
            this.diagnosticoClinicoSecundario = dados.diagnosticoClinicoSecundario.toUpperCase();
            this.procedimento = dados.procedimento.toUpperCase();
            this.cirurgiao = dados.cirurgiao.toUpperCase();

            this.auditado = dados.auditado;

            if (dados.obs !== undefined && dados.obs !== null) {
                this.obs = dados.obs.toUpperCase();
            } else {
                this.obs = "NENHUMA";
            }

            this.localProcedimento = dados.localProcedimento.toUpperCase();
            this.regiao = dados.regiao.toUpperCase();

            if (dados.grandeRegiao !== undefined && dados.grandeRegiao !== null) {
                this.grandeRegiao = dados.grandeRegiao.toUpperCase();
            }

            this.diametroMaior = dados.diametroMaior;
            this.diametroMenor = dados.diametroMenor;
            this.dataProcedimento = dados.dataProcedimento;

            // A imagem pode ser vazia
            dados.imagens !== undefined ?
                this.imagens = dados.imagens
                : this.imagens = new Array<Imagem>();

            // Dados para completar quando tiver o histopatologico

            if (dados.diagnosticoHisto !== undefined && dados.diagnosticoHisto !== null) {
                this.diagnosticoHisto = dados.diagnosticoHisto.toUpperCase();
            }

            if (dados.subtipoHisto !== undefined && dados.subtipoHisto !== null) {
                this.subtipoHisto = dados.subtipoHisto.toUpperCase();
            }


            if (dados.clark !== undefined && dados.clark !== null) {
                this.clark = dados.clark.toUpperCase();
            }

            this.breslow = dados.breslow;

            this.indiceMiotico = dados.indiceMiotico;

            if (dados.tipoTumor !== undefined && dados.tipoTumor !== null) {
                this.tipoTumor = dados.tipoTumor.toUpperCase();
            }

            this.tamanhoTumorDimMaior = dados.tamanhoTumorDimMaior;

            this.tamanhoTumorDimMenor = dados.tamanhoTumorDimMenor;

            if (dados.crescimentoRadial !== undefined && dados.crescimentoRadial !== null) {
                this.crescimentoRadial = dados.crescimentoRadial.toUpperCase();
            }

            if (dados.crescimentoVertical !== undefined && dados.crescimentoVertical !== null) {
                this.crescimentoVertical = dados.crescimentoVertical.toUpperCase();
            }

            if (dados.ulceracao !== undefined && dados.ulceracao !== null) {
                this.ulceracao = dados.ulceracao.toUpperCase();
            }

            if (dados.lesaoSatelite !== undefined && dados.lesaoSatelite !== null) {
                this.lesaoSatelite = dados.lesaoSatelite.toUpperCase();
            }

            if (dados.evidenciaRegressao !== undefined && dados.evidenciaRegressao !== null) {
                this.evidenciaRegressao = dados.evidenciaRegressao.toUpperCase();
            }

            if (dados.associacaoNevoMelanocito !== undefined && dados.associacaoNevoMelanocito !== null) {
                this.associacaoNevoMelanocito = dados.associacaoNevoMelanocito.toUpperCase();
            }

            if (dados.margemCirurgiaProfunda !== undefined && dados.margemCirurgiaProfunda !== null) {
                this.margemCirurgiaProfunda = dados.margemCirurgiaProfunda.toUpperCase();
            }

            if (dados.margemCirurgiaLateral !== undefined && dados.margemCirurgiaLateral !== null) {
                this.margemCirurgiaLateral = dados.margemCirurgiaLateral.toUpperCase();
            }

            if (dados.margemProfundaLivre !== undefined && dados.margemProfundaLivre !== null) {
                this.margemProfundaLivre = dados.margemProfundaLivre.toUpperCase();
            }

            if (dados.margemLateralLivre !== undefined && dados.margemLateralLivre !== null) {
                this.margemLateralLivre = dados.margemLateralLivre.toUpperCase();
            }

            if (dados.invasaoVascular !== undefined && dados.invasaoVascular !== null) {
                this.invasaoVascular = dados.invasaoVascular.toUpperCase();
            }

            if (dados.menorDistMargemRessecaoLateral !== undefined && dados.menorDistMargemRessecaoLateral !== null) {
                this.menorDistMargemRessecaoLateral = dados.menorDistMargemRessecaoLateral.toUpperCase();
            }

            if (dados.infiltracaoAngiolinfatica !== undefined && dados.infiltracaoAngiolinfatica !== null) {
                this.infiltracaoAngiolinfatica = dados.infiltracaoAngiolinfatica.toUpperCase();
            }

            if (dados.infiltracaoPerineural !== undefined && dados.infiltracaoPerineural !== null) {
                this.infiltracaoPerineural = dados.infiltracaoPerineural.toUpperCase();
            }

            if (dados.infiltradoLinfocito !== undefined && dados.infiltradoLinfocito !== null) {
                this.infiltradoLinfocito = dados.infiltradoLinfocito.toUpperCase();
            }

            if (dados.limiteRessecao !== undefined && dados.limiteRessecao !== null) {
                this.limiteRessecao = dados.limiteRessecao.toUpperCase();
            }

        } else {
            this.imagens = new Array<Imagem>();
        }
    }

    public getAuditado (): boolean {
        return this.auditado;
    }

    public setAuditado (auditado: boolean): void {
        this.auditado = auditado;
    }

    public getId(): number {
        return this.id;
    }

    public setId(id: number): void {
        this.id = id;
    }

    public getDiagnosticoClinico(): string {
        return this.diagnosticoClinico;
    }

    public setDiagnosticoClinico(diagnosticoClinico: string): void {
        this.diagnosticoClinico = diagnosticoClinico;
    }

    public getDiagnosticoClinicoSecundario(): string {
        return this.diagnosticoClinicoSecundario;
    }

    public setDiagnosticoClinicoSecundario(diagnosticoClinicoSecundario: string): void {
        this.diagnosticoClinicoSecundario = diagnosticoClinicoSecundario;
    }

    public getProcedimento(): string {
        return this.procedimento;
    }

    public setProcedimento(procedimento: string): void {
        this.procedimento = procedimento;
    }

    public getCirurgiao(): string {
        return this.cirurgiao;
    }

    public setCirurgiao(cirurgiao: string): void {
        this.cirurgiao = cirurgiao;
    }

    public getObs(): string {
        return this.obs;
    }

    public setObs(obs: string): void {
        this.obs = obs;
    }

    public getDiametroMaior(): number {
        return this.diametroMaior;
    }

    public setDiametroMaior(diametroMaior: number): void {
        this.diametroMaior = diametroMaior;
    }

    public getDiametroMenor(): number {
        return this.diametroMenor;
    }

    public setDiametroMenor(diametroMenor: number): void {
        this.diametroMenor = diametroMenor;
    }

    public getImagens(): Imagem[] {
        return this.imagens;
    }

    public setImagens(imagens: Imagem[]): void {
        this.imagens = imagens;
    }

    public getDataProcedimento(): Date {
        return this.dataProcedimento;
    }

    public setDataProcedimento(dataProcedimento: Date): void {
        this.dataProcedimento = dataProcedimento;
    }

    public getLocalProcedimento(): string {
        return this.localProcedimento;
    }

    public setLocalProcedimento(localProcedimento: string): void {
        this.localProcedimento = localProcedimento;
    }

    public getDiagnosticoHisto(): string {
        return this.diagnosticoHisto;
    }

    public setDiagnosticoHisto(diagnosticoHisto: string): void {
        this.diagnosticoHisto = diagnosticoHisto;
    }

    public getSubtipoHisto(): string {
        return this.subtipoHisto;
    }

    public setSubtipoHisto(subtipoHisto: string): void {
        this.subtipoHisto = subtipoHisto;
    }

    public getClark(): string {
        return this.clark;
    }

    public setClark(clark: string): void {
        this.clark = clark;
    }

    public getBreslow(): number {
        return this.breslow;
    }

    public setBreslow(breslow: number): void {
        this.breslow = breslow;
    }

    public getIndiceMiotico(): number {
        return this.indiceMiotico;
    }

    public setIndiceMiotico(indiceMiotico: number): void {
        this.indiceMiotico = indiceMiotico;
    }

    public getRegiao(): string {
        return this.regiao;
    }

    public setRegiao(regiao: string): void {
        this.regiao = regiao;
    }

    public setPacienteId (id: number) {
        this.pacienteId = 'paciente/' + id;
    }

    public getPacienteId (): string {
        return this.pacienteId;
    }

    public getLimiteRessecao(): string {
        return this.limiteRessecao;
    }

    public setLimiteRessecao(limiteRessecao: string): void {
        this.limiteRessecao = limiteRessecao;
    }

    public getInfiltracaoPerineural(): string {
        return this.infiltracaoPerineural;
    }

    public setInfiltracaoPerineural(infiltracaoPerineural: string): void {
        this.infiltracaoPerineural = infiltracaoPerineural;
    }

    public getInfiltracaoAngiolinfatica(): string {
        return this.infiltracaoAngiolinfatica;
    }

    public setInfiltracaoAngiolinfatica(infiltracaoAngiolinfatica: string): void {
        this.infiltracaoAngiolinfatica = infiltracaoAngiolinfatica;
    }

    public getInfiltradoLinfocito(): string {
        return this.infiltradoLinfocito;
    }

    public setInfiltradoLinfocito(infiltradoLinfocito: string): void {
        this.infiltradoLinfocito = infiltradoLinfocito;
    }


}
